jspsych-uil-session.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+if (typeof uil === 'undefined')
+    console.error("UiL Main library not loaded! " +
+                  "Refusing to load browser info functions");
+else
+{
+    if ('session' in uil) {
+        console.warn(
+            "uil.browser already exists, this is unexpected, " +
+                "we'll overwrite it."
+        );
+    }
+
+    uil.session = {};
+
+    (function (context) {
+        class API {
+            constructor(host, access_key) {
+                this.host = host;
+                this.headers = {};
+            }
+
+            _request(url, method, data) {
+                let params = {
+                    method: method,
+                };
+                if (typeof(data) !== 'undefined') {
+                    params.body = new Blob([data], {type: 'text/plain'});
+                }
+                return new Promise((resolve, reject) => {
+                    fetch(this.host + url, params).then((response) => {
+                        response.json().then(resolve).catch(reject);
+                    }).catch(reject);
+                });
+            }
+
+            get(url) {
+                return this._request(url, 'GET');
+            }
+            post(url, data) {
+                return this._request(url, 'POST', data);
+            }
```

Comment:
I think these methods should be marked private? Only exposing methods which correspond to API endpoints would be better imo.

Reply:
Do you like the python-style underscore convention or should it actually be inaccessible?

Reply:
Depends really; we have a lot of actually private methods in the utils, but I think in this case making a method private would result in some nasty JS tricks that don't improve code quality. The API class is semi-private already, so just using the underscore convention should do the job.

It's more of a hint to us, as the developers, that those methods are helper methods for the class and ideally should not be used outside of the class